### ISSUE TITLE
`React.forwardRef` type definition is broken in `@xyflow/react: 12.0.0-next.11`

### ISSUE THREAD

**Issue by snelsi** [xyflow/xyflow#4014]
### Describe the Bug

After upgrading to `@xyflow/react: 12.0.0-next.11` I started to see errors related to `forwardRef` and `displayName`:
![image](https://github.com/xyflow/xyflow/assets/15171036/eacb8423-f41c-4285-bee7-cafe902ac180)

This is a critical problem.

`packages/react/src/types/general.ts` contains this type declaration:

```tsx
// this is needed, to use generics + forwardRef
declare module 'react' {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function forwardRef<T, P = {}>(
    render: (props: P, ref: React.Ref<T>) => React.ReactElement | null
  ): (props: P & React.RefAttributes<T>) => React.ReactElement | null;
}
```

This code breaks proper `React.forwardRef` types and loses some of the extra props like `displayName` in the process.
I don't think modifying global type declarations in a library is a good practice.

### Steps to Reproduce the Bug or Issue

1. Upgrade to `@xyflow/react: 12.0.0-next.11`
2. Copy-paste this component into your app:

```tsx
import React from "react";

export interface ButtonProps {}

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(({ ...props }, ref) => <button {...props} ref={ref} />);

Button.displayName = "Button";
//           ^ Property 'displayName' does not exist on type '(props: ButtonProps& RefAttributes<HTMLButtonElement>) => ReactElement<any, string | JSXElementConstructor<any>> | null'.

export default React.memo(Button);
```

### Expected behavior

Don't modify global react type declarations.

### Platform

- OS: Windows 11
- Version: "@xyflow/react": "12.0.0-next.11"

**Comment by moklick** [comment]
yep! Good catch. We will change it in the next release!

**Comment by moklick** [comment]
This should be fixed in 12.0.0-next.12
